### DISCUSSION TITLE
Why UDP and Unix sockets have `take_error`, but not Tcp sockets?

### DISCUSSION THREAD

**Discussion by vi** [tokio-rs/tokio#3170] [Q&A] [answered]
https://docs.rs/tokio/0.3.4/tokio/net/struct.UnixListener.html?search=take_error

**Comment by taiki-e** [comment]
The direct reason `tokio` doesn't have that function is that [`mio` doesn't have it](https://docs.rs/mio/0.7.6/mio/?search=take_error).

I'm not a maintainer of `mio`, so I don't know the actual reason why `mio` doesn't have it, but I guess it's because `TcpSocket` was just added recently and no one ever needed that function. See also https://github.com/tokio-rs/tokio/issues/3082.

In any case, if you need that function, I recommend opening an issue (or a pr) in `mio`. Once added to `mio`, we can add it to `tokio` as well.

**Comment by taiki-e** [comment, marked as answer]
This will be added in https://github.com/tokio-rs/tokio/pull/4364.
